TypeConverter with List ModelClass won't pass build

I'm trying to create TypeCoverter with ModelClass List instance that looks like this:

```java
@com.raizlabs.android.dbflow.annotation.TypeConverter
public class IntListTypeConverter extends TypeConverter<String, ArrayList<Integer>>{

    @Override
    public String getDBValue(ArrayList<Integer> model) {
        String dbValue = "";
        for(int i : model) {
            dbValue += i + ";";
        }
        dbValue = dbValue.substring(0, dbValue.length() - 1);
        return dbValue;
    }

    @Override
    public ArrayList<Integer> getModelValue(String data) {
        ArrayList<Integer> modelValue = new ArrayList<>();
        for(String s : data.split(";")){
            modelValue.add(Integer.parseInt(s));
        }
        return modelValue;
    }
```

The error I'm getting is:
` Error:Execution failed for task ':app:compileStagingDebugJava'.
  java.lang.NullPointerException`

And here is last part of stacktrace for more information:
`Caused by: java.lang.NullPointerException
        at com.raizlabs.android.dbflow.processor.definition.TypeConverterDefinition.getModelClassQualifiedName(TypeConverterDefinition.java:60)
        at com.raizlabs.android.dbflow.processor.model.ProcessorManager.addTypeConverterDefinition(ProcessorManager.java:96)
        at com.raizlabs.android.dbflow.processor.handler.TypeConverterHandler.onProcessElement(TypeConverterHandler.java:43)
        at com.raizlabs.android.dbflow.processor.handler.BaseContainerHandler.handle(BaseContainerHandler.java:25)
        at com.raizlabs.android.dbflow.processor.model.ProcessorManager.handle(ProcessorManager.java:215)
        at com.raizlabs.android.dbflow.processor.DBFlowProcessor.process(DBFlowProcessor.java:85)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:793)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:722)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1700(JavacProcessingEnvironment.java:97)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1029)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1163)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1108)
        at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:824)
        at com.sun.tools.javac.main.Main.compile(Main.java:439)`

If I change ModelClass type to any other class build passes but i can't get it to work with List. Can you please help me?

--- a/README.md
+++ b/README.md
@@ -19,7 +19,7 @@ What sets this library apart: **every** feature has been unit tested to ensure f
 
 #### 1.4.2
 
-1. Throws error if the ```TypeConverterDefinition``` cannot process the type parameters, and warns devs of using type parameters for a ```TypeConverter```. Thanks [mariciv](https://github.com/mariciv)
+1. Throws error if the ```TypeConverterDefinition``` cannot process the type parameters, and warns devs of using type parameters for a typeparameter in a ```TypeConverter```. Thanks [mariciv](https://github.com/mariciv)
 2. Fixes ```BaseDatabaseDefinition.reset()``` fails to reset DB. Now appends ".db" to database name. Thanks [bafitor](https://github.com/bafitor)
 
 #### 1.4.1

--- a/README.md
+++ b/README.md
@@ -17,6 +17,10 @@ What sets this library apart: **every** feature has been unit tested to ensure f
 
 ## Changelog
 
+#### 1.4.2
+
+1. Throws error if the ```TypeConverterDefinition``` cannot process the type parameters, and warns devs of using type parameters for a ```TypeConverter```. Thanks [mariciv](https://github.com/mariciv)
+2. Fixes ```BaseDatabaseDefinition.reset()``` fails to reset DB. Now appends ".db" to database name. Thanks [bafitor](https://github.com/bafitor)
 
 #### 1.4.1
 

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 [![Android Weekly](http://img.shields.io/badge/Android%20Weekly-%23129-2CB3E5.svg?style=flat)](http://androidweekly.net/issues/issue-129)
 [![Android Arsenal](https://img.shields.io/badge/Android%20Arsenal-DBFlow-brightgreen.svg?style=flat)](https://android-arsenal.com/details/1/1134) 
-[![Raizlabs Repository](http://img.shields.io/badge/Raizlabs%20Repository-1.4.1-blue.svg?style=flat)](https://github.com/Raizlabs/maven-releases)
+[![Raizlabs Repository](http://img.shields.io/badge/Raizlabs%20Repository-1.4.2-blue.svg?style=flat)](https://github.com/Raizlabs/maven-releases)
 
 DBFlow
 ======
@@ -79,12 +79,12 @@ Add the library to the project-level build.gradle, using the [apt plugin](https:
 ```groovy
 
   dependencies {
-    apt 'com.raizlabs.android:DBFlow-Compiler:1.4.1'
-    aarLinkSources 'com.raizlabs.android:DBFlow-Compiler:1.4.1:sources@jar'
-    compile 'com.raizlabs.android:DBFlow-Core:1.4.1'
-    aarLinkSources 'com.raizlabs.android:DBFlow-Core:1.4.1:sources@jar'
-    compile 'com.raizlabs.android:DBFlow:1.4.1'
-    aarLinkSources 'com.raizlabs.android:DBFlow:1.4.1:sources@jar'
+    apt 'com.raizlabs.android:DBFlow-Compiler:1.4.2'
+    aarLinkSources 'com.raizlabs.android:DBFlow-Compiler:1.4.2:sources@jar'
+    compile 'com.raizlabs.android:DBFlow-Core:1.4.2'
+    aarLinkSources 'com.raizlabs.android:DBFlow-Core:1.4.2:sources@jar'
+    compile 'com.raizlabs.android:DBFlow:1.4.2'
+    aarLinkSources 'com.raizlabs.android:DBFlow:1.4.2:sources@jar'
   }
 
 ```

--- a/compiler/src/main/java/com/raizlabs/android/dbflow/processor/handler/TypeConverterHandler.java
+++ b/compiler/src/main/java/com/raizlabs/android/dbflow/processor/handler/TypeConverterHandler.java
@@ -7,6 +7,7 @@ import com.raizlabs.android.dbflow.converter.DateConverter;
 import com.raizlabs.android.dbflow.converter.SqlDateConverter;
 import com.raizlabs.android.dbflow.processor.definition.TypeConverterDefinition;
 import com.raizlabs.android.dbflow.processor.model.ProcessorManager;
+import com.raizlabs.android.dbflow.processor.validator.TypeConverterValidator;
 
 import java.util.Set;
 
@@ -18,6 +19,8 @@ import javax.lang.model.element.TypeElement;
  * adding default methods and adding them to the {@link com.raizlabs.android.dbflow.processor.model.ProcessorManager}
  */
 public class TypeConverterHandler extends BaseContainerHandler<TypeConverter> {
+
+    private static final TypeConverterValidator VALIDATOR = new TypeConverterValidator();
 
     private static final Class[] DEFAULT_TYPE_CONVERTERS = new Class[]{
             CalendarConverter.class,
@@ -40,6 +43,9 @@ public class TypeConverterHandler extends BaseContainerHandler<TypeConverter> {
 
     @Override
     protected void onProcessElement(ProcessorManager processorManager, Element element) {
-        processorManager.addTypeConverterDefinition(new TypeConverterDefinition((TypeElement) element, processorManager));
+        TypeConverterDefinition converterDefinition = new TypeConverterDefinition((TypeElement) element, processorManager);
+        if(VALIDATOR.validate(processorManager, converterDefinition)) {
+            processorManager.addTypeConverterDefinition(converterDefinition);
+        }
     }
 }

--- a/compiler/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.java
+++ b/compiler/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.java
@@ -1,0 +1,26 @@
+package com.raizlabs.android.dbflow.processor.validator;
+
+import com.raizlabs.android.dbflow.processor.definition.TypeConverterDefinition;
+import com.raizlabs.android.dbflow.processor.model.ProcessorManager;
+
+/**
+ * Description:
+ */
+public class TypeConverterValidator implements Validator<TypeConverterDefinition>  {
+    @Override
+    public boolean validate(ProcessorManager processorManager, TypeConverterDefinition typeConverterDefinition) {
+        boolean success = true;
+
+        if(typeConverterDefinition.getModelElement() == null) {
+            processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassElement().getSimpleName() +
+                " uses an unsupported Model Element parameter");
+            success = false;
+        } else if(typeConverterDefinition.getDbElement() == null) {
+            processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassElement().getSimpleName() +
+                    " uses an unsupported DB Element parameter");
+            success = false;
+        }
+
+        return success;
+    }
+}

--- a/compiler/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.java
+++ b/compiler/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.java
@@ -13,11 +13,13 @@ public class TypeConverterValidator implements Validator<TypeConverterDefinition
 
         if(typeConverterDefinition.getModelElement() == null) {
             processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassElement().getSimpleName() +
-                " uses an unsupported Model Element parameter");
+                " uses an unsupported Model Element parameter. If it has type parameters, you must remove them or subclass it" +
+                    "for proper usage.");
             success = false;
         } else if(typeConverterDefinition.getDbElement() == null) {
             processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassElement().getSimpleName() +
-                    " uses an unsupported DB Element parameter");
+                    " uses an unsupported DB Element parameter. If it has type parameters, you must remove them or subclass it " +
+                    "for proper usage.");
             success = false;
         }
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-VERSION_NAME=1.4.1
+VERSION_NAME=1.4.2
 VERSION_CODE=1
 GROUP=com.raizlabs.android
 

--- a/library/src/main/java/com/raizlabs/android/dbflow/config/BaseDatabaseDefinition.java
+++ b/library/src/main/java/com/raizlabs/android/dbflow/config/BaseDatabaseDefinition.java
@@ -108,6 +108,10 @@ public abstract class BaseDatabaseDefinition {
 
     public abstract String getDatabaseName();
 
+    public String getDatabaseFileName() {
+        return getDatabaseName() + ".db";
+    }
+
     public abstract int getDatabaseVersion();
 
     /**
@@ -132,7 +136,7 @@ public abstract class BaseDatabaseDefinition {
     public void reset(Context context) {
         if (!isResetting) {
             isResetting = true;
-            context.deleteDatabase(getDatabaseName());
+            context.deleteDatabase(getDatabaseFileName());
             mHelper = new FlowSQLiteOpenHelper(this, mInternalHelperListener);
             isResetting = false;
         }

--- a/library/src/main/java/com/raizlabs/android/dbflow/config/FlowSQLiteOpenHelper.java
+++ b/library/src/main/java/com/raizlabs/android/dbflow/config/FlowSQLiteOpenHelper.java
@@ -48,11 +48,11 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
     private SQLiteOpenHelper mBackupDB;
 
     public FlowSQLiteOpenHelper(BaseDatabaseDefinition flowManager, DatabaseHelperListener listener) {
-        super(FlowManager.getContext(), flowManager.getDatabaseName() + ".db", null, flowManager.getDatabaseVersion());
+        super(FlowManager.getContext(), flowManager.getDatabaseFileName(), null, flowManager.getDatabaseVersion());
         mListener = listener;
         mManager = flowManager;
 
-        movePrepackagedDB(getDatabaseFileName(), getDatabaseFileName());
+        movePrepackagedDB(mManager.getDatabaseFileName(), mManager.getDatabaseFileName());
 
         if (flowManager.backupEnabled()) {
             // Temp database mirrors existing
@@ -77,7 +77,7 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
                     executeMigrations(db, oldVersion, newVersion);
                 }
             };
-            restoreDatabase(getTempDbFileName(), getDatabaseFileName());
+            restoreDatabase(getTempDbFileName(), mManager.getDatabaseFileName());
             mBackupDB.getWritableDatabase();
         }
     }
@@ -88,11 +88,6 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
     private String getTempDbFileName() {
         return TEMP_DB_NAME + mManager.getDatabaseName() + ".db";
     }
-
-    private String getDatabaseFileName() {
-        return mManager.getDatabaseName() + ".db";
-    }
-
 
     /**
      * Pulled partially from code, runs the integrity check on pre-honeycomb devices.
@@ -145,7 +140,7 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
         // Try to copy database file
         try {
             // check existing and use that as backup
-            File existingDb = FlowManager.getContext().getDatabasePath(getDatabaseFileName());
+            File existingDb = FlowManager.getContext().getDatabasePath(mManager.getDatabaseFileName());
             InputStream inputStream;
             // if it exists and the integrity is ok
             if (existingDb.exists() && (mManager.backupEnabled() && FlowManager.isDatabaseIntegrityOk(mBackupDB))) {
@@ -254,7 +249,7 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
 
                 Context context = FlowManager.getContext();
                 File backup = context.getDatabasePath(getTempDbFileName());
-                File temp = context.getDatabasePath(TEMP_DB_NAME + "-2-" + getDatabaseFileName());
+                File temp = context.getDatabasePath(TEMP_DB_NAME + "-2-" + mManager.getDatabaseFileName());
 
                 // if exists we want to delete it before rename
                 if (temp.exists()) {
@@ -265,7 +260,7 @@ public class FlowSQLiteOpenHelper extends SQLiteOpenHelper {
                 if (backup.exists()) {
                     backup.delete();
                 }
-                File existing = context.getDatabasePath(getDatabaseFileName());
+                File existing = context.getDatabasePath(mManager.getDatabaseFileName());
 
                 try {
                     backup.getParentFile().mkdirs();

